ert_gui/simulation/run_dialog.py

```diff
@@ -200,31 +289,24 @@ def _on_ticker(self):
 
     @Slot(object)
     def _on_tracker_event(self, event):
-        if isinstance(event, GeneralEvent):
-            self.total_progress.setProgress(event.progress)
-            self.progress.setIndeterminate(event.indeterminate)
-
-            if event.indeterminate:
-                for state in event.sim_states:
-                    self.legends[state].updateLegend(state.name, 0, 0)
-            else:
-                for state in event.sim_states:
-                    try:
-                        self.progress.updateState(
-                            state.state, 100.0 * state.count / state.total_count)
-                    except ZeroDivisionError:
-                        # total_count not set by some slow tracker (EE)
-                        pass
-                    self.legends[state].updateLegend(
-                        state.name, state.count, state.total_count)
-
-        if isinstance(event, DetailedEvent):
-            if not self.progress.get_indeterminate():
-                self.detailed_progress.set_progress(event.details,
-                                                    event.iteration)
 
         if isinstance(event, EndEvent):
             self.simulation_done.emit(event.failed, event.failed_msg)
+            self._worker.stop()
+            self._ticker.stop()
+        elif isinstance(event, FullSnapshotEvent):
+            if event.snapshot is not None:
+                self._snapshot_model._add_snapshot(event.snapshot, event.iteration)
+            self._progress_view.setIndeterminate(event.indeterminate)
+            self._total_progress_bar.setValue(event.progress * 100)
+
+        elif isinstance(event, SnapshotUpdateEvent):
+            if event.partial_snapshot is not None:
+                self._snapshot_model._add_partial_snapshot(
+                    event.partial_snapshot, event.iteration
+                )
+            self._progress_view.setIndeterminate(event.indeterminate)
+            self._total_progress_bar.setValue(event.progress * 100)
```

Comment:
This triggers `DeprecationWarning: an integer is required (got type float).  Implicit conversion to integers using __int__ is deprecated, and may be removed in a future version of Python.` in Python 3.8. We should make the int conversion explicit.